vtex-apps/b2b-organizations
# Show the upload modal when the user selects bulk import

#### What problem is this solving?

This is the first PR adding the new **Bulk Import feature**. This includes:
1. Changing the button for creating organizations to be a menu with 2 options: creating a single org or importing in bulk.
2. Clicking the option for bulk importing the orgs will display an upload modal from [@vtex/bulk-import-ui](https://github.com/vtex/bulk-import-ui).

This PR **doesn't yet include**:
- Integration with the backend. It's currently mocking the request, simulating a 2 seconds-long successful call.
- The actual import process. For now we only handle the upload part of the process. A next PR will handle the call for when the user confirms that the uploaded file should be imported.
- Real translations. This will be done later by the localization team, in a separate PR that they'll create themselves.

#### How to test it?

[Workspace](https://mairabulk2--b2bstoreqa.myvtex.com/admin/b2b-organizations/organizations)

1. Click on the **Add Organization** menu.
2. Click on the "Import in Bulk" option.
3. Either click the **Upload** button to select a file, or drag/drop a file into the modal.

You should see a loading screen for ~2s, and then a success screen.

#### Screenshots or example usage:

<img width="1012" alt="Screenshot 2023-09-06 at 15 38 48" src="https://github.com/vtex-apps/b2b-organizations/assets/5216049/c6d8c9d0-37cd-457c-8a0f-d8a32270cf42">
<img width="1012" alt="Screenshot 2023-09-06 at 15 38 54" src="https://github.com/vtex-apps/b2b-organizations/assets/5216049/921ea5f1-521c-4632-ad41-549a1bcbbe34">
<img width="1012" alt="Screenshot 2023-09-06 at 15 39 16" src="https://github.com/vtex-apps/b2b-organizations/assets/5216049/9d444b5b-39c3-49b0-821b-0efc5ab9dd08">


#### How does this PR make you feel? [:link:](http://giphy.com/)



![](https://media.giphy.com/media/5zf2M4HgjjWszLd4a5/giphy.gif)
